Fixes TNL-363: EdX OpenID provider doesn't handle unicode e-mails

jbzdak opened the pull request:
## Components Affected 

* LMS 

## Discussion 

I've had problems with reproducing this issue manually, on my local vagrant instance, but the issue shows very well on unit tests. Problem is very easy to fix and shouldn't cause regressions --- just added ``u`` prefix to all audit log messages, as unicode variables are formatted into these log messages (values taken from ``request.POST`` are converted to unicode [by django][1], other values are taken from database --- and ``CharField`` returns unicode objects). 

Tests are long, because I decided to handle error cased by each of the audit log messages. 

## Legal

This PR is covered by OpenCraft contributor agreement.

@antoviaque

[1]: https://docs.djangoproject.com/en/1.8/ref/request-response/#django.http.QueryDict.__init__

edx-webhook commented:
Thanks for the pull request, @jbzdak! I've created [OSPR-992](https://openedx.atlassian.net/browse/OSPR-992) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).
We can't start reviewing your pull request until you've   submitted a [signed contributor agreement](http://open.edx.org/sites/default/files/wysiwyg/individual-contributor-agreement.pdf)   or indicated your institutional affiliation   and   added yourself to the [AUTHORS](https://github.com/edx/edx-platform/blob/master/AUTHORS) file . Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

sarina commented:
jenkins add to whitelist

sarina commented:
@jbzdak this looks good to me! Now we're waiting on the result of tests before I ask for a second reviewer. See https://openedx.atlassian.net/wiki/display/TE/Jenkins+Guide for more detail on how to understand build output.

jbzdak commented:
@sarina  I noticed errors in quality tests, some questions: 

* How can run these tests locally, ``paver run_quality`` displayed that everyting is OK? I have no experience whatsoever with EdX, so I might have done something wrong. 
* Should I change logging to use ``%s`` format, it's safe to do but really not related to the issue. 
* Most of the pylint errors seem fixable, apart of these two (see the [build output]): 

          15:33:27     293: E1101: (no-member), OpenIdProviderTest.test_provider_login_can_handle_unicode_email: 
          Instance of 'Sequence' has no 'encode' member
          15:33:27     333: E1101: (no-member), OpenIdProviderTest.test_provider_login_can_handle_unicode_email_inactive_account: 
          Instance of 'UserFactory' has no 'save' member

In the first case it seems that pylint incorrectly guessed the type, in second it assumed that ``UserFactory`` has no ``save`` (which is present). Should I just silence them? 

[1]: https://build.testeng.edx.org/job/edx-platform-quality-pr/9095/console

sarina commented:
See
https://github.com/edx/edx-platform/blob/master/docs/en_us/internal/testing.rst#python-code-style-quality
- sometimes running paver run_pylint first helps.

You can disable pylint warnings that are incorrect by using a pragma.

On Mon, Nov 30, 2015 at 10:53 AM, Jacek Bzdak <notifications@github.com>
wrote:

> @sarina <https://github.com/sarina> I noticed errors in quality tests,
> some questions:
>
>    - How can run these tests locally, paver run_quality displayed that
>    everyting is OK? I have no experience whatsoever with EdX, so I might have
>    done something wrong.
>    - Should I change logging to use %s format, it's safe to do but really
>    not related to the issue.
>    -
>
>    Most of the pylint errors seem fixable, apart of these two (see the
>    [build output]):
>
>      15:33:27     293: E1101: (no-member), OpenIdProviderTest.test_provider_login_can_handle_unicode_email:
>      Instance of 'Sequence' has no 'encode' member
>      15:33:27     333: E1101: (no-member), OpenIdProviderTest.test_provider_login_can_handle_unicode_email_inactive_account:
>      Instance of 'UserFactory' has no 'save' member
>
>
> In the first case it seems that pylint incorrectly guessed the type, in
> second it assumed that UserFactory has no save (which is present). Should
> I just silence them?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10761#issuecomment-160669106>.
>

jbzdak commented:
One more question: after I make the changes should I add new commits or just squash the changes?

sarina commented:
You'll need to squash before we merge but it makes reviews easier to wait
until the end.
On Nov 30, 2015 12:04 PM, "Jacek Bzdak" <notifications@github.com> wrote:

> One more question: after I make the changes should I add new commits or
> just squash the changes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10761#issuecomment-160690853>.
>

jbzdak commented:
Python tests passsed, should I be worried about the failing js/bokchoy/lettuce (I didn't touch any html or js), moreover on last commit they ran fine?

sarina commented:
eh probably just flaky tests. You can rerun them like so (I'll rerun them this time, but in the future you can):

jenkins run bokchoy
jenkins run js
jenkins run lettuce

jbzdak commented:
I'll try once again with bokchoy: 

jenkins run bokchoy

sarina commented:
@jbzdak I've asked the Sustaining team to take a look at your PR. You can follow the JIRA ticket to see discussion there. It may take a few days, especially as we all recover from the long US holiday last week.

sarina commented:
Bok-choy has been run 4 times on this PR and keep failing on 2 tests that I've filed as flaky: https://openedx.atlassian.net/browse/TNL-3852 and https://openedx.atlassian.net/browse/TNL-3853

I'm not running builds again; when the Ecom team reviews this PR, and you have to squash your commits, then can focus on getting the build passing.

jbzdak commented:
Thans for your help! I'll wait for the PR.

clintonb commented:
@jbzdak any reason not to use `from __future__ import unicode_literals`?

Otherwise, this looks good to me.

sarina commented:
@clintonb That is not a pattern we use in the platform, and I think some people internally actually have some opinions about that. Namely, it can get very confusing very fast when strings behave differently in different files. @nedbat  or @singingwolfboy likely have an opinion?

clintonb commented:
@sarina thanks.

:+1:

clintonb commented:
@jbzdak please squash your commits so that we can merge. Thanks for the contribution!

singingwolfboy commented on common/djangoapps/external_auth/tests/test_openid_provider.py:
```diff
@@ -260,27 +260,100 @@ def test_login_openid_handle_redirection_ratelimited(self):
 
+    def _attempt_login_and_perform_final_response(self, user, profile_name):
+        """
+        Performs full procedure of a succesfull OpenID provider login for user,
```
nitpick: "successful"

singingwolfboy commented on common/djangoapps/external_auth/tests/test_openid_provider.py:
```diff
@@ -260,27 +260,100 @@ def test_login_openid_handle_redirection_ratelimited(self):
+    def _attempt_login_and_perform_final_response(self, user, profile_name):
+        """
+        Performs full procedure of a succesfull OpenID provider login for user,
+        all required data is taken form ``user`` attrinute which is an instance
```
nitpick: "attribute"

singingwolfboy commented on common/djangoapps/external_auth/tests/test_openid_provider.py:
```diff
@@ -260,27 +260,100 @@ def test_login_openid_handle_redirection_ratelimited(self):
+        """
+        Performs full procedure of a succesfull OpenID provider login for user,
+        all required data is taken form ``user`` attrinute which is an instance
+        of ``User`` model. As a conveniance this method will also set
```
nitpick: "convenience"

singingwolfboy commented:
As far as `from __future__ import unicode_literals` goes, I agree with @sarina. It's not a pattern we use in the edx-platform repo, and it's confusing to have it implemented in some places but not others. The main two reasons to use that `__future__` import are either to progressively migrate to a fully-unicode program, one file at a time; or to make the code compatible with both Python 2 and Python 3 at the same time.

jbzdak commented:
Thank you for fast review, I have fixed spelling mistakes, and I have squashed this PR to two commits one for changes, another that adds me to ``AUTHORS`` file (this is what contributing guidelines instructed me to do).

clintonb commented:
@singingwolfboy press the merge button when you're ready.
